src/AzureClient/Magic/ConnectMagic.cs

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Jupyter.Core;
+using Microsoft.Quantum.IQSharp.Jupyter;
+
+namespace Microsoft.Quantum.IQSharp.AzureClient
+{
+    /// <summary>
+    ///     A magic command that can be used to connect to an Azure workspace.
+    /// </summary>
+    public class ConnectMagic : AzureClientMagicBase
+    {
+        private const string
+            ParamName_Login = "login",
+            ParamName_StorageAccountConnectionString = "storageAccountConnectionString",
+            ParamName_SubscriptionId = "subscriptionId",
+            ParamName_ResourceGroupName = "resourceGroupName",
+            ParamName_WorkspaceName = "workspaceName";
+
+        /// <summary>
+        ///     Constructs a new magic command given an IAzureClient object.
+        /// </summary>
+        public ConnectMagic(IAzureClient azureClient) :
+            base(azureClient,
+                "connect",
```

Comment:
Is there a disconnect? or do you call connect again if you want to connect to another workspace?

Reply:
Does connect return anything wen successful?
What errors can it return?
worth documenting these.

Reply:
Yes, you would just call connect again. We discussed disconnect but decided there wasn't really a scenario where it was needed.

---
In reply to: [426996900](https://github.com/microsoft/iqsharp/pull/144#discussion_r426996900) [](ancestors = 426996900)

Reply:
The return value of these magic commands is always an `ExecutionResult`, which indicates whether it succeeded and also (optionally) contains some information, either about the information that was retrieved, or about the error that occurred.

Yes, when switching these to all be subcommands of `%aq`, I'll need to revisit the return behavior and documentation for all of the commands to make sure they're consistent.

---
In reply to: [426997290](https://github.com/microsoft/iqsharp/pull/144#discussion_r426997290) [](ancestors = 426997290)

Reply:
I've opened a task to track this work.